src/scripts/validation.js

```diff
@@ -95,4 +95,90 @@ export const validationconfig = {
   errorClass: "modal__error_visible",
 };
 
-//enableValidation(settings);
+//enableValidation(settings);*/
+
+export const settings = {
+  formSelector: ".modal__form",
+  inputSelector: ".modal__input",
+  submitButtonSelector: ".modal__submit-btn",
+  inactiveButtonClass: "modal__submit-btn_disabled",
+  inputErrorClass: "modal__error",
+  errorClass: "modal__input-error_active",
+};
+
+const showInputError = (formElement, inputElement, errorMessage, config) => {
+  const errorMsgEl = formElement.querySelector(`#${inputElement.id}-error`);
+  errorMsgEl.textContent = errorMessage;
+  inputElement.classList.add(config.errorClass);
+};
+
+const hideInputError = (formElement, inputElement, config) => {
+  const errorMsgEl = formElement.querySelector(`#${inputElement.id}-error`);
+  errorMsgEl.textContent = "";
+  inputElement.classList.remove(config.errorClass);
+};
+
+const checkInputValidity = (formElement, inputElement, config) => {
+  if (!inputElement.validity.valid) {
+    showInputError(
+      formElement,
+      inputElement,
+      inputElement.validationMessage,
+      config
+    );
+  } else {
+    hideInputError(formElement, inputElement, config);
+  }
+};
+
+const hasInvalidInput = (inputList) => {
+  return inputList.some((element) => {
+    return !element.validity.valid;
+  });
+};
+
+const disableButton = (buttonElement, config) => {
+  buttonElement.classList.add(config.inactiveButtonClass);
+  buttonElement.disabled = true;
+};
+
+const toggleButtonState = (inputList, buttonElement, config) => {
+  if (hasInvalidInput(inputList)) {
+    disableButton(buttonElement, config);
+  } else {
+    buttonElement.classList.remove(config.inactiveButtonClass);
+    buttonElement.disabled = false;
+  }
+};
+
+const setEventListeners = (formElement, config) => {
+  const inputList = Array.from(
+    formElement.querySelectorAll(config.inputSelector)
+  );
+  const buttonElement = formElement.querySelector(config.submitButtonSelector);
+  toggleButtonState(inputList, buttonElement, config);
+  formElement.addEventListener("submit", () => {
```

Comment:
`submit` should be changed to `reset` here because you need to **disable** the button only after resetting the form (not right after submission)




```js
  formElement.addEventListener("reset", () => {
    disableButton(buttonElement, config);
  });

```

Reply:
* My remark hasn't been corrected https://github.com/lstapleton02/se_project_spots/pull/2#discussion_r2050500455 


`submit` should be changed to `reset` here because you need to **disable** the button only after resetting the form (not right after submission)




```js
  formElement.addEventListener("reset", () => {
    disableButton(buttonElement, config);
  });

```